frontend/src/features/admin-form/create/logic/BuilderLogicContext.tsx

```diff
@@ -0,0 +1,87 @@
+import { createContext, FC, useContext, useMemo, useState } from 'react'
+
+import { FormFieldDto } from '~shared/types/field'
+import { LogicDto } from '~shared/types/form'
+
+import { useAdminForm } from '~features/admin-form/common/queries'
+
+export type FormFieldWithQuestionNumber = FormFieldDto & {
+  questionNumber: number
+}
+
+type BuilderLogicContextReturn = {
+  /**
+   * Map of field _id to form field for easy retrieval of form data when
+   * rendering logic.
+   */
+  mapIdToField: Record<FormFieldDto['_id'], FormFieldWithQuestionNumber>
+  /**
+   * The array of form logic in the form.
+   */
+  formLogics?: LogicDto[]
+  /**
+   * ID of currently active logic. Used to highlight logic in the form, or to
+   * disable certain components.
+   */
+  activeLogicId?: LogicDto['_id']
+  /**
+   * Whether a new logic block should be rendered and other actions disabled.
+   */
+  hasPendingLogic?: boolean
+  /**
+   * Set the active logic ID when a logic block is about to be manipulated.
+   */
+  handleSetActiveLogicId: (id: LogicDto['_id']) => void
+  /**
+   * Set whether a new logic block should be rendered and other actions disabled.
+   */
+  handleSetHasPendingLogic: (hasLogic: boolean) => void
```

Comment:
might we be taking a performance hit here by using Context instead of a store? Context will cause all consumers to rerender when the value changes, whereas a store allows us to subscribe to slices of the state